viewer/src/dataModels/pointCloud/internal/PointCloudMetadataRepository.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { ModelUrlProvider } from '@/utilities/networking/types';
+import { EptSceneProvider } from './EptSceneProvider';
+import { PointCloudMetadata } from '@/dataModels/pointCloud/public/PointCloudMetadata';
+
+// TODO 25-05-2020 j-bjorne: Move into common type folder
+import { DataRepository } from '@/dataModels/cad/internal/DataRepository';
+import { CadTransformationProvider } from '@/dataModels/cad/internal/CadTransformationProvider';
+
+export class PointCloudMetadataRepository<TModelIdentifier>
+  implements DataRepository<TModelIdentifier, Promise<PointCloudMetadata>> {
+  private readonly _modelMetadataProvider: ModelUrlProvider<TModelIdentifier> & EptSceneProvider;
+  private readonly _cadTransformationProvider: CadTransformationProvider;
+  constructor(
+    modelBlobProvider: ModelUrlProvider<TModelIdentifier> & EptSceneProvider,
```

Comment:
What is this ` & EptSceneProvider` trickery? Should create a type alias for this imo

Reply:
Like, you want me to add a
```type ModelMetadataProvider<TModelIdentifier> = ModelUrlProvider<TModelIdentifier> & EptSceneProvider;```

to the file?

Reply:
Yes (seems strange that that type which doesn't consists of two types not mentioning Metadata turns into a MetadataProvider though)

Reply:
Those two parts are what makes up the metadata. I didn't want to make it a metadataDataProvider, but we could name it that.

Reply:
`    modelBlobProvider: ModelUrlProvider<TModelIdentifier> & EptSceneProvider,` - is this a `PointCloudMetadataProvider`? Need to make the names here consistent.